packages/components/src/utils/validation.ts

```diff
@@ -21,6 +21,28 @@ export const REF_HREF_PATTERN =
 export const REF_INTERNAL_HREF_PATTERN =
   `(${ALLOWED_URL_REGEXES.internal})|(${ALLOWED_URL_REGEXES.legacy})` as const
 
+// Validation for form-related embed URLs
+export const isValidFormSGEmbedUrl = (url: string) => {
+  if (!url) {
+    return false
+  }
+
+  try {
+    const urlObject = new URL(url)
+    return urlObject.hostname === "form.gov.sg"
+  } catch (_) {
+    return false
+  }
+}
+
+export const FORMSG_EMBED_URL_REGEXES = {
+  formsg: "^https://form\\.gov\\.sg/[a-z0-9]*$",
+} as const
```

Comment:
should be `FORMS_` not `FORMSG_`

Reply:
Hmm tbh this one gets a bit weird, cos the component is specifically for FormSG but our map/video components are more generic so the pattern we use here is more suitable for generic ones. For now I think we are not planning to embed other types of forms (e.g. Typeform) so hence decided to go with specifically FormSG here to standardise the naming

Reply:
yeah though since we have `fomsg` in the key, i thought it will be more consistent to have it be `FORMS_`, to standardize with MAPS and VIDEOS, otherwise feels slightly weird to do `FORMSG_EMBED_URL_REGEXES`.formsg`

Reply:
anyway i approve first, this is not really a blocker for me